DisguiseUnityRenderStream/Runtime/Util/AutoDisposable.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+
+#if UNITY_EDITOR
+using UnityEditor;
+#else
+using UnityEngine;
+#endif
+
+namespace Disguise.RenderStream.Utils
+{
+    abstract class AutoDisposable : IDisposable
+    {
+        protected AutoDisposable()
+        {
+            Register();
+        }
+
+        void Register()
+        {
+#if UNITY_EDITOR
+            EditorApplication.quitting += Dispose;
+            AssemblyReloadEvents.beforeAssemblyReload += Dispose;
+#else
+            Application.quitting += Dispose;
+#endif
+        }
+
+        public virtual void Dispose()
```

Comment:
This method can be made `abstract`. This class does not need to inherit `IDisposable`.